### DISCUSSION TITLE
Having issue when Test Cases Running on Code which is using read-replica.

### DISCUSSION THREAD

**Discussion by masterdungeon** [pytest-dev/pytest#11802] [Q&A]
I have an API, for which I am writing test cases. that API used read-replica to fetch the data.

`MyModel.objects.using('replica').all().count()` this is the code which is written in API.
Now when test cases running it returns 0. but when I am modifying this to `MyModel.objects..all().count()` then it returns 1.

We are creating a new blank database for running test cases every time.

**Comment by RonnyPfannschmidt** [comment]
i presume you are using django?

im not familiar with the api you use there and absolutely zero details are rpovided about your deployment, dependencies and setup, so we cannot give any advice in good conscience 

i recommend updating the question with details on the used libraries, test integrations and setup so we have the glimpse of a chance to help

**Reply by masterdungeon** [reply]
Hi,

Yes, I am using django with pytest for test cases execution and currently i am just running test cases in my local.

Test Cases Setup::

```
TEST_DATABASE = 'test_' + os.environ['DATABASE_NAME']

def pytest_configure(config):
    """
    called before test process is started.
    """
    '''creating db dump of schema'''
    command = "pg_dump -U {} -p {} -h {} -p 5432 -s {}  > db.sql".format(
        os.environ['DATABASE_USER'],
        os.environ['DATABASE_PASSWORD'],
        os.environ['DATABASE_HOST'],
        os.environ['DATABASE_NAME']
    )
    os.system(command)

    '''Dropping existing database if any'''
    run_sql('DROP DATABASE IF EXISTS ' + TEST_DATABASE)

    '''Creating database'''
    run_sql("create database " + TEST_DATABASE + ";")

    '''Dumping schema'''
    os.system("psql -U "+ os.environ['DATABASE_USER'] +" -d " + TEST_DATABASE + " < db.sql")
    os.system("rm -rf db.sql")


def pytest_unconfigure(config):
    """
    called before test process is exited.
    """
    print("Dropping database")
    connections.close_all()
    run_sql("drop database " + TEST_DATABASE + ";")
 

@pytest.fixture(scope='session')
def django_db_setup(django_db_setup, django_db_blocker):
    with django_db_blocker.unblock():
        create_data() # this methods create basic data like users and other master data.
```

Dependencies:

```
pytest==4.6.11
pytest-django==3.10.0
```

Code which is being tested:

```
def records_data(user_id):
    user = User.objects.get(id=user_id)
    records = MyModel.objects.using('replica').available(user)
    return records
```

when above code executes it gives me 0 records.
but when i fire `records = MyModel.objects.available(user)`, it gives me 1 record.

Note: We have created fixtures to create data for MyModel.
